api/src/pcapi/core/mails/transactional/pro/offer_validation_to_pro.py

```diff
@@ -67,21 +67,27 @@ def retrieve_data_for_validated_offer_rejection_email(
     )
 
 
-def send_offer_validation_status_update_email(
+def get_email_data_from_offer(
     offer: Offer | educational_models.CollectiveOffer | educational_models.CollectiveOfferTemplate,
     old_status: OfferValidationStatus,
     validation_status: OfferValidationStatus,
-    recipient_emails: list[str],
-) -> None:
+) -> models.TransactionalEmailData:
     if validation_status is OfferValidationStatus.APPROVED:
         offer_data = retrieve_data_for_offer_approval_email(offer)
-        mails.send(recipients=recipient_emails, data=offer_data)
+        return offer_data
+
+    # REJECTED
+    if old_status is OfferValidationStatus.PENDING:
+        offer_data = retrieve_data_for_pending_offer_rejection_email(offer)
+    elif old_status is OfferValidationStatus.APPROVED:
+        offer_data = retrieve_data_for_validated_offer_rejection_email(offer)
+    else:
+        offer_data = retrieve_data_for_offer_rejection_email(offer)
+    return offer_data
```

Comment:
Suggestion : tu peux suivre la même logique de retour immédiat, comme tu as fait au-dessus.
```suggestion
    if old_status is OfferValidationStatus.PENDING:
        return retrieve_data_for_pending_offer_rejection_email(offer)
    if old_status is OfferValidationStatus.APPROVED:
        return retrieve_data_for_validated_offer_rejection_email(offer)
    return retrieve_data_for_offer_rejection_email(offer)
```